ResearchKit/Consent/ORKEAGLMoviePlayerView.m

```diff
@@ -580,14 +662,18 @@ - (BOOL)compileShader:(GLuint *)shader type:(GLenum)type URL:(NSURL *)URL {
     
     glGetShaderiv(*shader, GL_COMPILE_STATUS, &status);
     if (status == 0) {
+        [self restoreGLContext];
```

Comment:
Can we rearrange this slightly to avoid the early return? Better to have a single exit path.